src/settlement-chain/NodeRegistry.sol

```diff
@@ -105,41 +90,44 @@ contract NodeRegistry is INodeRegistry, Migratable, ERC721Upgradeable {
 
     /// @inheritdoc INodeRegistry
     function initialize() public initializer {
-        __ERC721_init("XMTP Node Operator", "XMTP");
+        __ERC721_init("XMTP Nodes", "nXMTP");
 
         // TODO: This should probably come from the parameter registry.
         _getNodeRegistryStorage().maxCanonicalNodes = 20;
     }
 
-    /* ============ Admin-Only Functions ============ */
+    /* ============ Admin Functions ============ */
 
     /// @inheritdoc INodeRegistry
     function addNode(
-        address to_,
-        bytes calldata signingKeyPub_,
-        string calldata httpAddress_,
-        uint256 minMonthlyFee_
-    ) external onlyAdmin returns (uint256 nodeId_) {
-        require(to_ != address(0), InvalidAddress());
-        require(signingKeyPub_.length > 0, InvalidSigningKey());
+        address owner_,
+        bytes calldata signingPublicKey_,
+        string calldata httpAddress_
+    ) external onlyAdmin returns (uint32 nodeId_, address signer_) {
+        require(owner_ != address(0), InvalidOwner());
+        require(signingPublicKey_.length > 0, InvalidSigningPublicKey());
         require(bytes(httpAddress_).length > 0, InvalidHttpAddress());
 
         NodeRegistryStorage storage $ = _getNodeRegistryStorage();
 
+        require((uint256($.nodeCount) + 1) * NODE_INCREMENT <= type(uint32).max, MaxNodesReached());
+
         unchecked {
-            nodeId_ = uint256(++$.nodeCount) * NODE_INCREMENT; // The first node starts with `nodeId_ = NODE_INCREMENT`.
+            nodeId_ = ++$.nodeCount * NODE_INCREMENT; // The first node starts with `nodeId_ = NODE_INCREMENT`.
         }
 
+        signer_ = _toAddress(keccak256(signingPublicKey_));
+
         // Nodes start off as non-canonical.
-        $.nodes[nodeId_] = Node(signingKeyPub_, httpAddress_, false, minMonthlyFee_);
+        $.nodes[nodeId_] = Node(signer_, false, signingPublicKey_, httpAddress_);
 
-        _mint(to_, nodeId_);
+        _mint(owner_, nodeId_);
 
-        emit NodeAdded(nodeId_, to_, signingKeyPub_, httpAddress_, minMonthlyFee_);
+        emit NodeAdded(nodeId_, owner_, signer_, signingPublicKey_, httpAddress_);
     }
 
     /// @inheritdoc INodeRegistry
-    function addToNetwork(uint256 nodeId_) external onlyAdmin {
+    function addToNetwork(uint32 nodeId_) external onlyAdmin {
```

Comment:
@mkysel had the request of making `addToNetwork` reentrant, meaning that calling addToNetwork twice won't fail the second time with `NodeAlreadyInCanonicalNetwork()`.

This is not an action item as it's currently handled in the xmtpd-cli (which will parse for specific errors and drop them silently), but leaving the comment here in case it makes sense to do something about it at contract level as well.